lib/sles4sap/sap_deployment_automation_framework/networking.pm

```diff
@@ -0,0 +1,389 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::sap_deployment_automation_framework::networking;
+
+use strict;
+use warnings;
+use testapi;
+use Exporter qw(import);
+use Carp qw(croak);
+use Regexp::Common qw(net);
+use Time::Piece;
+use sles4sap::azure_cli
+  qw(az_storage_blob_list az_storage_blob_lease_acquire az_storage_blob_upload az_storage_blob_update
+  az_network_peering_list az_network_vnet_get);
+
+=head1 SYNOPSIS
+
+Library with common functions for Microsoft SDAF deployment automation related to networking setup.
+
+=cut
+
+our @EXPORT = qw(
+  assign_address_space
+  calculate_subnets
+);
+
+=head2 calculate_net_addr_space
+
+    calculate_net_addr_space();
+
+Calculate network IP space that will be reserved to contain all underlying subnets: 'admin', 'db', 'app', 'web'.
+Each network space will reserve 64 IP addresses. This splits last IP octet into 4 network spaces.
+First 64 address spaces are used from third octet, making 256 network spaces available for all OpenQA tests at
+any point in time. (256 tests can run at the same time using one control plane.)
+
+=cut
+
+sub calculate_net_addr_space {
+    # map { $_ * 64 IP addresses } (1 .. all 256 addresses within octet/64 IP per network space);
+    my @addr_ranges = map { $_ * 64 } (1 .. 256 / 64);
```

Comment:
Why `1 .. 256 / 64` instead of `1 .. 4`? Or the intention is for this to return `1/64, 2/64, 3/64, .... 256/64`, because if it's the second, then it needs some parameters. `/` will take precedence over `..`

Reply:
It's the first one. The final array is `64, 128, 192,  256`. 
I was just trying to make it self explanatory how did I get to the number 4.
I'll change it to 4 since it has comment above explaining it .